src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -1345,6 +1445,11 @@ private void AssignPatternVariables(BoundPattern pattern)
         public override BoundNode VisitBlock(BoundBlock node)
         {
             DeclareVariables(node.Locals);
+
+            // First assemble possibly-unassigned reads for all
+            // local functions
+            RecordReadsInLocalFunctions(node);
```

Comment:
>RecordReadsInLocalFunctions [](start = 12, length = 27)

Data flow analysis may require visiting a block multiple times. But we should only record reads once.